### DISCUSSION TITLE
Rust-analyzer in vs-code is failing to fetch "cargo metadata"

### DISCUSSION THREAD

**Discussion by sanamsarath** [rust-lang/rust-analyzer#14387] [Q&A]
Hi,

I am new to rust and still figuring out how things work. I am trying to setup rust dev environment using vs-code and rust-analyzer extension. 

In my dev environment I can't access public crates.io and has a config file defined with "crates-io-mirror" (proxy registry) as a replacement for the "crates.io" Config file also includes the registry index and token.

Cargo tool from terminal works fine, but for some reason the rust-analyzer extension in the vs-code is failing. Any kind of debugging pointers are appreciated?? like how to run the rust-analyzer binary from terminal?? any specific debug arguments to pass etc.,

Error log
```
[ERROR rust_analyzer::main_loop] FetchWorkspaceError:
rust-analyzer failed to load workspace: Failed to read Cargo metadata from Cargo.toml file /xxxxxx/sd_monitor/Cargo.toml, Some(Version { major: 1, minor: 65, patch: 0 }): Failed to run `cd "/xxxxxxx/sd_monitor" && "cargo" "metadata" "--format-version" "1" "--features" "" "--manifest-path" "/xxxxxx/sd_monitor/Cargo.toml" "--filter-platform" "x86_64-unknown-linux-gnu"`: `cargo metadata` exited with an error: error: failed to get `hyper` as a dependency of package `sd_monitor v0.1.0 (/xxxxxx/sd_monitor)`

Caused by:
  failed to load source for dependency `hyper`

Caused by:
  Unable to update registry `crates-io`

Caused by:
  could not find a configured source with the name `crates-io-mirror` when attempting to lookup `crates-io` (configuration in `/xxxxxx/sd_monitor/.cargo/config.toml`)
```

**Comment by bjorn3** [comment]
What is the content of `/xxxxxx/sd_monitor/.cargo/config.toml`? You can censor all url's if you want.

**Reply by sanamsarath** [reply]
@bjorn3  Attached is the screen shot of the config file
![image](https://user-images.githubusercontent.com/13935372/226757325-d79b9059-09ee-4ad6-b1a7-bc60fbc614fa.png)

**Reply by bjorn3** [reply]
That should be fine. Can you try running `cd "/xxxxxxx/sd_monitor" && "cargo" "metadata" "--format-version" "1" "--features" "" "--manifest-path" "/xxxxxx/sd_monitor/Cargo.toml" "--filter-platform" "x86_64-unknown-linux-gnu"`? That is the command rust-analyzer says it tried to run.

**Reply by sanamsarath** [reply]
That command ran successfully from terminal and the output is a large set of data

**Reply by sanamsarath** [reply]
The root cause for my issue was some default buildvm script in the ~/.profile that hogs the session/terminal start for 60s. This is causing buildenvironment related issues in the vs-code extension host. Disabling the script fixed things for me.

The error logs look like below
```
2023-03-22 17:57:19.274 [error] ExtensionHostConnection#buildUserEnvironment resolving shell environment failed {}
2023-03-22 17:57:19.276 [error] ExtensionHostConnection#buildUserEnvironment resolving shell environment failed {}
2023-03-22 17:57:19.277 [error] ExtensionHostConnection#buildUserEnvironment resolving shell environment failed {}
2023-03-22 17:57:19.277 [error] ExtensionHostConnection#buildUserEnvironment resolving shell environment failed {}
2023-03-22 17:57:43.423 [error] ExtensionHostConnection#buildUserEnvironment resolving shell environment failed {}
```
